server/reload.go

```diff
@@ -1294,31 +1294,70 @@ func (s *Server) reloadOCSP() error {
 	s.mu.Unlock()
 
 	// Stop all OCSP Stapling monitors in case there were any running.
-	var wasEnabled bool
 	for _, oc := range ocsps {
-		wasEnabled = true
 		oc.stop()
 	}
 
+	type ocspState struct {
+		config     *tls.Config
+		wasEnabled bool
+	}
+	withOCSP := make(map[string]ocspState)
+
+	// Start OCSP Stapling for client connections.
+	if config := sopts.TLSConfig; config != nil {
```

Comment:
These guards look like they're requiring OCSP staples if _any_ options are enabled for TLS for these items, rather than if OCSP is enabled.  Am I misreading?

Reply:
Good point, we could use a better name for this variable. The check whether OCSP Stapling is enabled or not does not happen until `NewOCSPMonitor` is called and checks each one of the certificates.

Reply:
Okay, as long as it's a naming issue and not a logic flaw, then I'll go ahead and approve.